internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -730,6 +811,40 @@ func (b *BeatsMonitor) getHttpStreams(
 	return httpStreams
 }
 
+// getPrometheusStream returns the stream definition for prometheus/metrics input.
+// Note: The return type must be []any due to protobuf serialization quirks.
+func (b *BeatsMonitor) getPrometheusStream(
+	failureThreshold *uint,
+	metricsCollectionIntervalString string,
+) any {
+	prometheusHost := b.getCollectorTelemetryEndpoint()
+
+	// want metricset "collector"
+	monitoringNamespace := b.monitoringNamespace()
+	indexName := fmt.Sprintf("metrics-elastic_agent.collector-%s", monitoringNamespace)
+	dataset := "elastic_agent.collector"
```

Comment:
You need https://github.com/elastic/elastic-agent/issues/8970 (this issue specifies an OTLP formatted data stream as is which would require the prometheus receiver) for this to be writable by Fleet managed agents I think. Right now I would expect the agent API key from Fleet to not have permissions to create or index to this data stream.